File: algorithms/1_easy/M/03090_maximum_length_substring_with_two_occurrences/sliding_window.py
class Solution:
    def maximumLengthSubstring(self, s: str) -> int:
        n = len(s)
        freqs = [0] * 256
        lo = 0
        res = 0

        for hi in range(n):
            freqs[ord(s[hi])] += 1

            while freqs[ord(s[hi])] > 2:
                freqs[ord(s[lo])] -= 1
                lo += 1

            res = max(res, hi + 1 - lo)

        return res

File: algorithms/1_easy/M/03090_maximum_length_substring_with_two_occurrences/sliding_window.rs
impl Solution {
    pub fn maximum_length_substring(s: String) -> i32 {
        let bytes = s.as_bytes();
        let len = bytes.len();
        let mut freqs = [0_i8; 256];
        let mut lo = 0_usize;
        let mut max = 0_usize;

        for hi in 0..len {
            freqs[bytes[hi] as usize] += 1;

            while freqs[bytes[hi] as usize] > 2 {
                freqs[bytes[lo] as usize] -= 1;
                lo += 1;
            }

            max = max.max(hi + 1 - lo);
        }

        max.try_into().expect("int overflow")
    }
}
